engine/utils/collections/BitSet.h

```diff
@@ -0,0 +1,435 @@
+//
+//
+//  Copyright (c) 2022 Jonathan Moallem (@J-Mo63) & Aryeh Zinn (@Raelr)
+//
+//  This code is released under an unmodified zlib license.
+//  For conditions of distribution and use, please see:
+//      https://opensource.org/licenses/Zlib
+//
+
+#ifndef SIEGE_ENGINE_BITSET_H
+#define SIEGE_ENGINE_BITSET_H
+
+namespace Siege::Utils
+{
+/**
+ * A static class containing bit manipulation utilities. This class contains universal functions for
+ * setting, unsetting, and managing complex bitmasks.
+ */
+class BitUtils
+{
+public:
+
+    /**
+     * A set of enums representing the exact positions of specific bits in a byte based on their
+     * index position + 1.
+     */
+    enum BIT_POSITIONS
+    {
+        BIT_POSITION_1 [[maybe_unused]] = 1,
+        BIT_POSITION_2 [[maybe_unused]] = 2,
+        BIT_POSITION_4 [[maybe_unused]] = 3,
+        BIT_POSITION_8 [[maybe_unused]] = 4,
+        BIT_POSITION_16 [[maybe_unused]] = 5,
+        BIT_POSITION_32 [[maybe_unused]] = 6,
+        BIT_POSITION_64 [[maybe_unused]] = 7,
+        BIT_POSITION_128 [[maybe_unused]] = 8,
```

Comment:
We don't actually apply `[[maybe_unused]]` anywhere - I'm ok with leaving it be?

Reply:
this is mostly because Clion is a pain and keeps spamming me about this. These enums are definiitely in use but CLion isn't smart enough to detect this.